tests/components/bluetooth/test_base_scanner.py

```diff
@@ -394,3 +395,113 @@ async def test_restore_history_remote_adapter(hass, hass_storage):
+    hass, caplog, enable_bluetooth
+):
+    """Test a device with a 10 minute advertising interval."""
+    manager = _get_manager()
+
+    bparasite_device = BLEDevice(
+        "44:44:33:11:23:45",
+        "bparasite",
+        {},
+        rssi=-100,
+    )
+    bparasite_device_adv = generate_advertisement_data(
+        local_name="bparasite",
+        service_uuids=[],
+        manufacturer_data={1: b"\x01"},
+        rssi=-100,
+    )
+
+    class FakeScanner(BaseHaRemoteScanner):
+        def inject_advertisement(
+            self, device: BLEDevice, advertisement_data: AdvertisementData
+        ) -> None:
+            """Inject an advertisement."""
+            self._async_on_advertisement(
+                device.address,
+                advertisement_data.rssi,
+                device.name,
+                advertisement_data.service_uuids,
+                advertisement_data.service_data,
+                advertisement_data.manufacturer_data,
+                advertisement_data.tx_power,
+                {"scanner_specific_data": "test"},
+            )
+
+    new_info_callback = manager.scanner_adv_received
+    connector = (
+        HaBluetoothConnector(MockBleakClient, "mock_bleak_client", lambda: False),
+    )
+    scanner = FakeScanner(hass, "esp32", "esp32", new_info_callback, connector, False)
+    unsetup = scanner.async_setup()
+    cancel = manager.async_register_scanner(scanner, True)
+
+    monotonic_now = time.monotonic()
+    new_time = monotonic_now
+    bparasite_device_went_unavailable = False
+
+    @callback
+    def _bparasite_device_unavailable_callback(_address: str) -> None:
+        """Bparasite device unavailable callback."""
+        nonlocal bparasite_device_went_unavailable
+        bparasite_device_went_unavailable = True
+
+    advertising_interval = 60 * 10
+
+    bparasite_device_unavailable_cancel = bluetooth.async_track_unavailable(
+        hass,
+        _bparasite_device_unavailable_callback,
+        bparasite_device.address,
+        connectable=False,
+    )
+
+    for _ in range(0, 20):
+        new_time += advertising_interval
+        with patch(
+            "homeassistant.components.bluetooth.base_scanner.MONOTONIC_TIME",
+            return_value=new_time,
+        ):
+            scanner.inject_advertisement(bparasite_device, bparasite_device_adv)
+
+    future_time = new_time
+    assert (
+        bluetooth.async_address_present(hass, bparasite_device.address, False) is True
+    )
+    assert bparasite_device_went_unavailable is False
+    with patch(
+        "homeassistant.components.bluetooth.manager.MONOTONIC_TIME",
+        return_value=new_time,
+    ):
+        async_fire_time_changed(hass, dt_util.utcnow() + timedelta(seconds=future_time))
+        await hass.async_block_till_done()
+
+    assert bparasite_device_went_unavailable is False
+
+    missed_advertisement_future_time = (
+        future_time + FALLBACK_MAXIMUM_STALE_ADVERTISEMENT_SECONDS + 1
```

Comment:
This is wrong. It should be the advertising interval but if I change the test it will fail